src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -956,6 +956,53 @@ def copy_external_global(self, globalvar, tag=None):
                         " tag '{1}' is already used by another symbol."
                         "".format(globalvar, tag))
 
+    def rename_symbol(self, symbol, name):
+        '''
+        Rename the given symbol which should belong to this symbol table
+        with the new name provided.
+
+        :param symbol: the symbol to be renamed.
+        :type symbol: :py:class:`psyclone.psyir.symbols.Symbol`
+        :param str name: the new name.
+
+        :raises TypeError: if the symbol is not a Symbol.
+        :raises TypeError: if the name is not a str.
+        :raises ValueError: if the given symbol does not belog to this symbol \
+                            table.
+        :raises ValueError: if the given variable name already exists in the \
+                            symbol table.
+
+        '''
+        if not isinstance(symbol, Symbol):
+            raise TypeError(
+                "The symbol argument of rename_symbol() must be a Symbol, but"
+                " found: '{0}'.".format(type(symbol).__name__))
+
+        if symbol not in self.symbols:
+            raise ValueError(
+                "The symbol argument of rename_symbol() must belong to this "
+                "symbol_table instance, but '{0}' does not.".format(symbol))
+
+        if not isinstance(name, str):
+            raise TypeError(
+                "The name argument of rename_symbol() must be a str, but"
+                " found: '{0}'.".format(type(symbol).__name__))
+
+        if name in self._symbols:
+            raise ValueError(
+                "The name argument of rename_symbol() must not already exist "
+                "in this symbol_table instance, but '{0}' does.".format(name))
+
+        # Delete current dictionary entry
+        del self._symbols[symbol.name]
+
+        # Rename symbol using protected access as symbol cannot expose a setter
```

Comment:
When you say "cannot", do you mean "should not"?

Reply:
Yes, udpated.